recourse_methods/mrmc_method.py

```diff
@@ -92,47 +92,85 @@ def alpha(dist: np.ndarray) -> np.ndarray:
 
 
 class MRM:
-    """Monotone Recourse Measures generates recourse directions."""
+    """Monotone Recourse Measures (MRM) generates recourse directions."""
 
     def __init__(
         self,
         dataset: pd.DataFrame,
         adapter: recourse_adapter.RecourseAdapter,
         alpha: AlphaFunction = get_volcano_alpha(),
-        rescale_direction: RecourseRescaler = normalize_rescaler,
+        rescale_direction: Optional[RecourseRescaler] = normalize_rescaler,
+        confidence_threshold: Optional[float] = None,
+        model: Optional[model_interface.Model] = None,
+        _processed_data: Optional[recourse_adapter.EmbeddedDataFrame] = None,
     ):
         """Creates an MRM instance.
 
+        The dataset is processed for recourse using MRM.process_data(). This
+        processing step can be skipped by providing recourse data directly via
+        _processed_data. This is typically unnecessary for end users.
+
+        Data provided to _processed_data should
+            * Consist only of positively classified examples
+            * Not have the label column
+            * Be transformed into an embedded numerical space by the adapter.
+
         Args:
             dataset: The dataset to provide recourse over.
             adapter: A RecourseAdapter object to transform the data between
                 human-readable and embedded space.
             alpha: The alpha function to use during recourse generation.
             rescale_direction: A function for rescaling the recourse.
+            confidence_threshold: If provided, MRM only generates directions
+                pointing to areas of sufficiently high model confidence.
+            model: Used to check per-datapoint model confidence if
+                confidence_threshold is given.
+            _processed_data: A processed, recourse-ready dataset to use instead
+                of the provided dataset.
         """
-        self.data = MRM.process_data(dataset, adapter)
+        if _processed_data is not None:
+            self.data = _processed_data
+        else:
+            self.data = MRM._process_data(
+                dataset,
+                adapter,
+                confidence_threshold=confidence_threshold,
+                model=model,
+            )
         self.adapter = adapter
         self.alpha = alpha
         self.rescale_direction = rescale_direction
 
     @staticmethod
-    def process_data(
+    def _process_data(
         dataset: pd.DataFrame,
         adapter: recourse_adapter.RecourseAdapter,
+        confidence_threshold: Optional[float] = None,
+        model: Optional[model_interface.Model] = None,
     ) -> recourse_adapter.EmbeddedDataFrame:
         """Processes the dataset for MRM.
-        It filters out negative outcomes and transforms the data to a numeric
-        embedded space.
+
+        It filters out negative outcome points, drops the class label, and
```

Comment:
Suggestion for explicitness and clarity: "Removes datapoints with negative (-1) labels, drops the class label feature column, ... Also excludes datapoints with below-threshold model confidence."